experiments/01-first-scenario.md

```diff
@@ -0,0 +1,256 @@
+# Table of contents
+- [GitOps tools setup](#gitops-tools-setup)
+  - [Install Argo CD](#install-argo-cd)
+  - [Install Flux CD](#install-flux-cd)
+- [Deploying guestbook application](#deploying-guestbook-application)
+  - [Fork argocd-example-apps repository](#fork-argocd-example-apps-repository)
+  - [Deploy the application using Argo CD](#deploy-the-application-using-argo-cd)
+  - [Deploy the application using Flux CD](#deploy-the-application-using-flux-cd)
+  - [Performing a rolling update](#performing-a-rolling-update)
+  - [Performing a rollback](#performing-a-rollback)
+  - [Clean up](#clean-up)
+- [Deploying podinfo application](#deploying-podinfo-application)
+  - [Fork podinfo repository](#fork-podinfo-repository)
+  - [Set the initial version](#set-the-initial-version)
+  - [Deploy the application using Argo CD](#deploy-the-application-using-argo-cd-1)
+  - [Deploy the application using Flux CD](#deploy-the-application-using-flux-cd-1)
+  - [Performing a rolling update](#performing-a-rolling-update-1)
+  - [Performing a rollback](#performing-a-rollback-1)
+  - [Clean up](#clean-up-1)
+
+# GitOps tools setup
+
+## Install Argo CD
+```bash
+kubectl create namespace argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+#### Login to Argo CD
+```bash
+# start a port-forward
+kubectl port-forward svc/argocd-server -n argocd 8080:443
+
+# Get the initial password in another window
+PASSWORD=$(argocd admin initial-password -n argocd | head -n 1 | awk '{print $NF}')
+
+# Login to ArgoCD
+argocd login localhost:8080 --insecure --username=admin --password=$PASSWORD
+
+# (Optional) Check the clusters
+argocd cluster list
+```
+
+## Install Flux CD
+```bash
+kubectl apply -f https://github.com/fluxcd/flux2/releases/latest/download/install.yaml
+```
+
+# Deploying guestbook application
+
+## Fork argocd-example-apps repository
+```bash
+gh repo fork https://github.com/argoproj/argocd-example-apps.git
+export GITHUB_USERNAME="YOUR_GITHUB_USERNAME"
+```
+
+## Deploy the application using Argo CD
+```bash
+kubectl create namespace guestbook-argocd
+argocd app create guestbook \
+  --app-namespace argocd \
```

Comment:
got this error here: `Error: unknown flag: --app-namespace`

Reply:
I didn't this error, could it be related to CLI version?
`--dest-namespace` is the namespace where the actual application, guestbook, gets deployed, while `--app-namespace` is where the object, Application, gets created.

Reply:
could be! which version of argocd are you using?

Reply:
I am using v2.6.7+5bcd846